base/special/rem_pio2.jl

```diff
@@ -0,0 +1,305 @@
+# This file is a part of Julia. Except for the rem_pio2_kernel, and
+# cody_waite_* methods (see below) license is MIT: https://julialang.org/license
+
+# rem_pio2_kernel and cody_waite_* methods are heavily based on FDLIBM code:
+# __ieee754_rem_pio2, that is made available under the following licence:
+
+## Copyright (C) 1993 by Sun Microsystems, Inc. All rights reserved.
+##
+## Developed at SunPro, a Sun Microsystems, Inc. business.
+## Permission to use, copy, modify, and distribute this
+## software is freely granted, provided that this notice
+## is preserved.
+
+const invpio2 =  6.36619772367581382433e-01
+const pio2_1  =  1.57079632673412561417e+00
+const pio2_1t =  6.07710050650619224932e-11
+const pio2_2  =  6.07710050630396597660e-11
+const pio2_2t =  2.02226624879595063154e-21
+const pio2_3  =  2.02226624871116645580e-21
+const pio2_3t =  8.47842766036889956997e-32
+
+"""
+    highword(x)
+
+Return the high word of `x` as a `UInt32`.
+"""
+@inline highword(x::UInt64) = unsafe_trunc(UInt32,x >> 32)
+@inline highword(x::Float64) = highword(reinterpret(UInt64, x))
+
+function cody_waite_2c_pio2(x::Float64)
+    # The three lines below could be replaced with:
+    #     fn = round(x*invpio2, RoundNearestTiesUp)
+    # at a small cost
+    fn = x*invpio2+6.755399441055744e15
+    fn = fn-6.755399441055744e15
+    n  = Int32(fn)
+    cody_waite_2c_pio2(x, fn, n)
+end
+function cody_waite_2c_pio2(x, fn, n)
+    z = x - fn*pio2_1
```

Comment:
`muladd` should be fine here as well (since `pio2_1` was chosen so that it would be exact)